server/services/omnichannel-voip/service.ts

```diff
@@ -17,12 +20,63 @@ export class OmnichannelVoipService extends ServiceClass implements IOmnichannel
 
 	private users: UsersRaw;
 
+	private voipRoom: VoipRoomsRaw;
+
 	constructor(db: Db) {
 		super();
 		this.users = new UsersRaw(db.collection('users'));
+		this.voipRoom = new VoipRoomsRaw(db.collection('rocketchat_room'));
 		this.logger = new Logger('OmnichannelVoipService');
 	}
 
+	private async createVoipRoom(
+		rid: string,
+		name: string,
+		agent: { agentId: string; username: string },
+		guest: ILivechatVisitor,
+	): Promise<string> {
+		const status = 'online';
+		const { _id, username, department: departmentId } = guest;
+		const newRoomAt = new Date();
+
+		this.logger.debug(`Creating Voip room for visitor ${_id}`);
+
+		const room = Object.assign({
+			_id: rid,
+			msgs: 0,
+			usersCount: 1,
+			lm: newRoomAt,
+			fname: name,
+			t: 'v',
+			ts: newRoomAt,
+			departmentId,
+			v: {
+				_id,
+				username,
+				token: guest.token,
+				status,
+			},
+			servedBy: {
+				_id: agent.agentId,
+				ts: newRoomAt,
+				username: agent.username,
+			},
+			cl: false,
+			open: true,
+			waitingResponse: true,
+			// this should be overriden by extraRoomInfo when provided
+			// in case it's not provided, we'll use this "default" type
+			source: {
+				type: 'voip',
+			},
+			queuedAt: newRoomAt,
+
+			// We assume room is created when call is started (there could be small delay)
+			callStarted: newRoomAt,
+		});
```

Comment:
```suggestion
		const room = {
			_id: rid,
			msgs: 0,
			usersCount: 1,
			lm: newRoomAt,
			fname: name,
			t: 'v',
			ts: newRoomAt,
			departmentId,
			v: {
				_id,
				username,
				token: guest.token,
				status,
			},
			servedBy: {
				_id: agent.agentId,
				ts: newRoomAt,
				username: agent.username,
			},
			cl: false,
			open: true,
			waitingResponse: true,
			// this should be overriden by extraRoomInfo when provided
			// in case it's not provided, we'll use this "default" type
			source: {
				type: 'voip',
			},
			queuedAt: newRoomAt,

			// We assume room is created when call is started (there could be small delay)
			callStarted: newRoomAt,
		};
```

If this gives you an error you may need to fix something on the IRoom definition.